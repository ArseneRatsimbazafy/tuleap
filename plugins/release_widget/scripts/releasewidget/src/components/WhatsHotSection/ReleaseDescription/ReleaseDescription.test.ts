/*
 * Copyright (c) Enalean, 2019 - present. All Rights Reserved.
 *
 * This file is a part of Tuleap.
 *
 * Tuleap is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * Tuleap is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Tuleap. If not, see <http://www.gnu.org/licenses/>.
 */

import { shallowMount, ShallowMountOptions, Wrapper } from "@vue/test-utils";
import ReleaseDescription from "./ReleaseDescription.vue";
import { createStoreMock } from "../../../../../../../../src/www/scripts/vue-components/store-wrapper-jest";
import { MilestoneData, StoreOptions } from "../../../type";
import { createReleaseWidgetLocalVue } from "../../../helpers/local-vue-for-test";

let releaseData: MilestoneData & Required<Pick<MilestoneData, "planning">>;
const component_options: ShallowMountOptions<ReleaseDescription> = {};
const project_id = 102;

describe("ReleaseDescription", () => {
    let store_options: StoreOptions;
    let store;

    async function getPersonalWidgetInstance(
        store_options: StoreOptions
    ): Promise<Wrapper<ReleaseDescription>> {
        store = createStoreMock(store_options);

        component_options.mocks = { $store: store };
        component_options.localVue = await createReleaseWidgetLocalVue();

        return shallowMount(ReleaseDescription, component_options);
    }

    beforeEach(() => {
        store_options = {
            state: {}
        };

        releaseData = {
            id: 2,
            planning: {
                id: "100"
            },
            number_of_artifact_by_trackers: []
        };

        component_options.propsData = {
            releaseData
        };
    });

    it("Given user display widget, Then a good link to top planning of the release is rendered", async () => {
        store_options.state.project_id = project_id;

        const wrapper = await getPersonalWidgetInstance(store_options);
        expect(wrapper.element).toMatchSnapshot();
    });
});
