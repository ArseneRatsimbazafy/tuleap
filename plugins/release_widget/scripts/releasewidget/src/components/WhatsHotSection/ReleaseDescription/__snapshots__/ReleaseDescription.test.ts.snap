// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReleaseDescription Given user display widget, Then a good link to top planning of the release is rendered 1`] = `
<div
  class="release-content-description"
>
  <release-description-badges-tracker-stub
    release-data="[object Object]"
    releasedata="[object Object]"
  />
   
  <div
    class="release-description-row"
  >
    <div
      class="release-description"
    />
     
    <a
      data-test="overview-link"
      href="/plugins/agiledashboard/?group_id=102&planning_id=100&action=show&aid=2&pane=details"
    >
      <i
        class="release-description-link-icon fa fa-long-arrow-right"
      />
       
      <translate-stub
        tag="span"
      >
         Go to release overview 
      </translate-stub>
    </a>
  </div>
   
  <div
    class="release-chart-burndown-row"
  >
    <h2
      class="tlp-pane-subtitle"
      data-current-language="en_US"
      data-msgid="Burndown"
    >
      Burndown
    </h2>
     
    <p
      class="empty-pane-text"
      data-current-language="en_US"
      data-msgid="There is nothing here!"
    >
      There is nothing here!
    </p>
  </div>
</div>
`;
